test/integration/scheduler/framework_test.go

```diff
@@ -2279,26 +2159,53 @@ func initTestSchedulerForFrameworkTest(t *testing.T, testCtx *testutils.TestCont
 }
 
 // initRegistryAndConfig returns registry and plugins config based on give plugins.
-// TODO: refactor it to a more generic functions that accepts all kinds of Plugins as arguments
-func initRegistryAndConfig(t *testing.T, pp ...*PermitPlugin) (frameworkruntime.Registry, schedulerconfig.KubeSchedulerProfile) {
-	var registry frameworkruntime.Registry
+func initRegistryAndConfig(t *testing.T, pp ...framework.Plugin) (frameworkruntime.Registry, schedulerconfig.KubeSchedulerProfile) {
 	if len(pp) == 0 {
 		return frameworkruntime.Registry{}, schedulerconfig.KubeSchedulerProfile{}
 	}
 
+	registry := frameworkruntime.Registry{}
+	plugins := &v1beta3.Plugins{}
+
+	for _, p := range pp {
+		// register plugin
+		switch pl := p.(type) {
+		case *PermitPlugin:
```

Comment:
I don't quite like this switch. What about accepting the builder instead?

```golang
func initRegistryAndConfig(t *testing.T, plBuilder ...framework.Builder)
```

Reply:
I moved this logic to `newPlugin` and removed other plugin initialization functions, any advices?